htdocs/sendtoken.php

```diff
@@ -48,7 +49,26 @@
 if (isset($_REQUEST["login"]) and $_REQUEST["login"]) { $login = strval($_REQUEST["login"]);}
 else { $result = "loginrequired";}
 
-if (! isset($_POST["mail"]) and ! isset($_REQUEST["login"])) { $result = "emptysendtokenform"; }
+if ( $result === "" and ( ! isset($_REQUEST["formtoken"]) or ! $_REQUEST["formtoken"] )  ) {
+    $result = "missingformtoken";
+}
+
+if (! isset($_POST["mail"]) and ! isset($_REQUEST["login"])) {
+
+    $result = "emptysendtokenform";
+
+    # Generate formtoken
+    ini_set("session.use_cookies",0);
+    ini_set("session.use_only_cookies",1);
+    ini_set("session.use_strict_mode",0);
```

Comment:
We use `ini_set` in a lot of places in our code, and here in this page.

Maybe we should move them to index.php?

Reply:
Good point!

Maybe at some point we would need to have cookies sent for saving the session at client side. So, I am rather thinking to a dedicated function.

Also I am not quite satisfied by calling ini_set, whereas:
1. there are some warning in debug mode about misusage of ini_set function
2. it seems you can set those parameters in the session_start function (but my tests don't work for now)

Reply:
Yes, using php session mechanism was an easy way to deal with tokens, but we may find a better solution. It will require a token backend on server side, which should be something easy to set up (files, in memory base, ...)

Reply:
Unfortunately, I don't see a simple and clean way for doing this. Indeed, in `lib/captcha/InternalCaptcha.php` we also use ini_set, but in this context we need cookies.

I think the best would be to find and use a php lib for managing server sessions. I am opening a dedicated issue for that. Not sure we can fix this for next 1.7.0 release though.